### ISSUE TITLE
Database keeps clearing out for some unknown reason.

### ISSUE THREAD

**Issue by cherrerajobs** [almarklein/timetagger#443]
I'm running a local server in Python 3.10 I've have my records for the month cleared out two times now. I don't know exactly what causes it. I'm on a guest account on localhost. This time it happened I did the following.

1. Had many records for the month
2. I accidently opened up a second tab to the timetagger page on localhost while one was clocking some time
3. I stopped the clock on one of the tabs
4. I refreshed one of the tabs
5. I closed out one of the tabs, the data on the still open tab still had my records.
6. I put the computer to sleep
7. Woke the computer up the next day and the timetagger server was no longer running on wakeup and it looked like the page had gotten to a login page before the server closed. I assume that happened in the process of putting the computer to sleep the day before.

How do I keep timetagger from totally erasing my database lol. Now I'm guessing how much time I spent.

**Comment by almarklein** [comment]
@cherrerajobs I can understand that this is very annoying. From what you explain in your post I don't see an obvious reason of what's going on, but I can give a view pointers to check.

The server losing it's records is very strange. There is no logic in the server where it e.g. deletes a database. This means that either the database got deleted by something else, or it never stored the records in the first place.

The browser application stores all records locally, this is so that when your wifi is down, you can just keep tracking, and when the wifi connection is up again, it will sync all changes to the browser. However, all this local data is deleted when you log out or clear your browser cache.

* If you are running the server in a Docker container, you have to make sure to map the directory containing the databases to a directory on the host system. Otherwise all is lost when the container restarts.
* If you are running the timetagger server locally, with Python on your machine, be aware that the directory where the databases are stored is writable by the process, and is not a temporary directory. Linux clears its temp dirs on each reboot.
* You could check the database file, and see if it changes size when you create a new record
* And finally, in the browser app, make sure that the sync icon next to the menu-icon does not show an error. Otherwise the changes you make never reach the server and get deleted when logging out.

I hope this helps you resolve your problem.
----

<img width="238" alt="image" src="https://github.com/almarklein/timetagger/assets/3015475/71b28b58-6847-4edb-bbe6-fde4d6ddc785">

*Sync icon indicating a connection problem. You can click it to get more details.*

**Comment by cherrerajobs** [comment]
Thank you for the quick response.

I did notice that error indicator periodically actually, I didn't realise I could click it, I would hover over it but nothing would show. A `cursor: pointer` style would be a nice addition there for desktop. This was on Windows by the way.

I think the browser cache explains it as I would periodically close and restart my browser and even open up new tabs to the local timetagger server and it would fetch all the records I had created without a hitch. My .db file hasn't been updated since December 15th 2023 though.

I can confirm it's a caching confusion as I can load the timetagger app on the browser but not on a private browser/tab (The server is not actually running).

So what REALLY happened was:

1. I started Timetagger
2. Recorded many records over time without the server even running.
3. Restarted the computer and went to the url of the local server
4. Timetagger loads up normally so I assume it's all working. Meanwhile the little error indicator is trying to tell me it can't connect to the server because it's not actually running
5. Eventually clearing out the browser cache destroys the local cache.

Knowing that I now have the understanding so that it won't happen again.

I think the following information box can be even more useful if it explained a little bit more about local browser caching and the server:

![image](https://github.com/almarklein/timetagger/assets/8894763/2149e1f8-8fc3-477a-a293-b88d7900fc6a)

Or if the error message did instead:
![image](https://github.com/almarklein/timetagger/assets/8894763/db5791e9-4ca2-456e-81bc-ce3202e94737)

I'm sure I missed all this info in my eagerness to get it all up and running. Thanks again.
